src/hatch/types.py

```diff
@@ -0,0 +1,8 @@
+from __future__ import annotations
+
+# from os import PathLike as _PathLike
+from typing import Union
+
+from hatch.utils.fs import Path
+
+PathLike = Union[Path, str]
```

Comment:
Is there a reason not to use `os.PathLike` and `os.fspath` instead of `str(path)`?